api/post.go

```diff
@@ -470,11 +473,11 @@ func getOpenGraphMetadata(c *Context, w http.ResponseWriter, r *http.Request) {
 	og := app.GetOpenGraphMetadata(url)
 
 	ogJSON, err := og.ToJSON()
+	openGraphDataCache.AddWithExpiresInSecs(props["url"], ogJSON, openGraphDataCacheExpireTime) // Cache would expire after 1 week
```

Comment:
website preview seldom changes, I don't think we need to send requests once an hour.

I moved this line before error check to put an empty og data into cache, in this way mm wouldn't request outside every time this func invoked just because the site doesn't support og format, then waiting for another failure.